crates/chain/tests/common/tx_template.rs

```diff
@@ -0,0 +1,138 @@
+use std::collections::HashMap;
+
+use bdk_chain::{
+    indexed_tx_graph::IndexedTxGraph, keychain::KeychainTxOutIndex, ConfirmationHeightAnchor,
+};
+use bitcoin::{
+    hashes::Hash, locktime::absolute::LockTime, OutPoint, ScriptBuf, Sequence, Transaction, TxIn,
+    TxOut, Txid, Witness,
+};
+
+/// Transaction template.
+pub struct TxTemplate<'a, K, A> {
+    /// Uniquely identifies the transaction, before it can have a txid.
+    pub tx_name: &'a str,
+    pub inputs: &'a [TxInTemplate<'a>],
+    pub outputs: &'a [TxOutTemplate<K>],
+    pub anchors: &'a [A],
+    pub last_seen: Option<u64>,
+}
+
+impl<'a, K, A> Default for TxTemplate<'a, K, A> {
+    fn default() -> Self {
+        Self {
+            tx_name: Default::default(),
+            inputs: Default::default(),
+            outputs: Default::default(),
+            anchors: Default::default(),
+            last_seen: Default::default(),
+        }
+    }
+}
+
+impl<'a, K, A> Copy for TxTemplate<'a, K, A> {}
+
+impl<'a, K, A> Clone for TxTemplate<'a, K, A> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+#[allow(dead_code)]
+pub enum TxInTemplate<'a> {
+    /// This will give a random txid and vout.
+    Bogus,
+
+    /// This is used for coinbase transactions because they do not have previous outputs.
+    Coinbase,
+
+    /// Contains the `tx_name` and `vout` that we are spending. The rule is that we must only spend
+    /// a previous transaction.
+    PrevTx(&'a str, usize),
+}
+
+pub struct TxOutTemplate<K> {
+    pub value: u64,
+    pub owned_spk: Option<(K, u32)>, // some = derive a spk from K, none = random spk
+}
+
+#[allow(dead_code)]
+pub fn init_graph<'a>(
+    graph: IndexedTxGraph<ConfirmationHeightAnchor, KeychainTxOutIndex<()>>,
```

Comment:
Nit: I think it will be clearer if we just had a `descriptor` input, and not construct an `IndexedTxGraph` from an `IndexedTxGraph`.